enterprise/backup/src/main/java/org/neo4j/backup/ConsistencyCheck.java

```diff
@@ -0,0 +1,121 @@
+                }
+            },
+    EXPERIMENTAL
+            {
+                @Override
+                public boolean runFull( File storeDir, Config tuningConfiguration,
+                        ProgressMonitorFactory progressFactory, LogProvider logProvider,
+                        FileSystemAbstraction fileSystem, PageCache pageCache, boolean verbose )
+                        throws ConsistencyCheckFailedException
+                {
+                    try
+                    {
+                        return new org.neo4j.consistency.ConsistencyCheckService().runFullConsistencyCheck(
+                                storeDir, tuningConfiguration, progressFactory,
+                                logProvider, fileSystem, pageCache, verbose ).isSuccessful();
+                    }
+                    catch ( org.neo4j.consistency.checking.full.ConsistencyCheckIncompleteException e )
+                    {
+                        throw new ConsistencyCheckFailedException( e );
+                    }
+                }
+            },
+    LEGACY
+            {
+                @Override
+                public boolean runFull( File storeDir, Config tuningConfiguration,
+                        ProgressMonitorFactory progressFactory, LogProvider logProvider,
+                        FileSystemAbstraction fileSystem, PageCache pageCache, boolean verbose )
+                        throws ConsistencyCheckFailedException
+                {
+                    try
+                    {
+                        return new org.neo4j.legacy.consistency.ConsistencyCheckService().runFullConsistencyCheck(
+                                storeDir, tuningConfiguration, progressFactory,
+                                logProvider, fileSystem, pageCache ).isSuccessful();
+                    }
+                    catch ( org.neo4j.legacy.consistency.checking.full.ConsistencyCheckIncompleteException e )
+                    {
+                        throw new ConsistencyCheckFailedException( e );
+                    }
+                }
+            };
+
+    public abstract boolean runFull( File storeDir, Config tuningConfiguration,
+            ProgressMonitorFactory progressFactory, LogProvider logProvider,
+            FileSystemAbstraction fileSystem, PageCache pageCache, boolean verbose )
+            throws ConsistencyCheckFailedException;
+
+    public static ConsistencyCheck fromString( String name )
+    {
+        for ( ConsistencyCheck consistencyCheck : values() )
+        {
+            if ( consistencyCheck.toString().equalsIgnoreCase( name ) )
+            {
+                return consistencyCheck;
+            }
+        }
+        throw new IllegalArgumentException( "Unknown consistency check name: " + name +
+                                            ". Supported values: " + Arrays.toString( values() ) );
+    }
+
+    public static class ConsistencyCheckFailedException extends Exception
```

Comment:
I've just realised that is slightly strange to have exception as inner class of the enum. Can we please make it top level package-private class?